### PR TITLE
making c1visual a test execution, and c1manual validation

### PR DESCRIPTION
**Pull request by dczulada**

### CODE REVIEW

**Comment by dczulada** [comment]
When A test is first created 

screen shot 2016-07-06 at 8 48 46 am

Partially filled out form, submit is disabled

screen shot 2016-07-06 at 8 50 10 am

In progress after Uploading a file

screen shot 2016-07-06 at 8 52 50 am

Test Results

screen shot 2016-07-06 at 8 54 04 am

**Comment by thecristen** [comment]
9 first-level bullets = a really long list that will not get read. They seem like they could be split up - maybe the last 4 bullets could go with box #2? And bullet #1 ("Fill in the Data Criteria Checklist") could be the title of box #1!

File: app/views/checklist_tests/_checklist_measures.html.erb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  <%= bootstrap_nested_form_for([product, product_test], :url => { :controller => 'checklist_tests', :action => 'update' }) do |f| %>
+    <% measures.each do |measure| %>
+      <div class = 'panel-group' id = '<%= measure.cms_id %>' disabled = true>
+        <div class = 'panel panel-default'>
+          <div class = 'panel-heading'>
+            <h1 class='panel-title'>
+              <%= "#{measure.cms_id} #{measure.name}" %>
+            </h1>
+          </div>
+          <div class = 'panel checklist-panel'>
```
Comment (thecristen) [inline app/views/checklist_tests/_checklist_measures.html.erb]: there should only be one `.panel`, which wraps the `.panel-heading` and `.panel-body` and `.panel-footer`, so this one I think is extraneous and probably what's adding the excessive padding..

File: app/views/checklist_tests/show.html.erb [hunk 2]
```diff
@@ -1,112 +1,42 @@
...
-              <%= f.submit 'Save', class: 'btn btn-default' %>
-              <span class = 'small-indent'>this saves the entire checklist page</span>
-            </div>
-          </div>
-        </div>
+  <div class = 'test-steps'>
+    <div class="panel panel-info">
+      <div class="panel-heading">
+        <h1 class="panel-title test-step">
+          <%= 1 %> <i class="fa fa-fw fa-bolt" aria-hidden="true"></i> <%= 'Checklist Instructions' %>
```
Comment (thecristen) [inline app/views/checklist_tests/show.html.erb]: 1 and 'Checklist Instructions' don't need to be evaluated as if there are Ruby...

File: app/views/checklist_tests/show.html.erb [hunk 3]
```diff
@@ -1,112 +1,42 @@
...
+        </h1>
       </div>
-    <% end %>
-  <% end %>
-<% end %>
-
+      <div class="panel-body">
+        <%= render partial: 'checklist_instructions', locals: { instructions: APP_CONFIG['tests']['ChecklistTest']['instructions'] } %>
```
Comment (thecristen) [inline app/views/checklist_tests/show.html.erb]: `checklist_instructions` also has a `.panel-body`, so this creates a `.panel-body` nested within a `.panel-body`.

File: app/views/checklist_tests/show.html.erb [hunk 4]
```diff
@@ -1,112 +1,42 @@
...
+      </div>
+    </div>
+  </div>
+  <div class = 'panel-body' id = 'display_checklist_measures' >
```
Comment (thecristen) [inline app/views/checklist_tests/show.html.erb]: This probably doesn't need to be a `.panel-body`. FWIW `.panel-body` doesn't really do anything if it's not inside of a `.panel`.

File: app/views/checklist_tests/_checklist_measures.html.erb [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+              <table class = 'table table-sm'>
+                <thead>
+                  <tr>
+                    <th class = 'col-sm-2'>Validated in QRDA</th>
+                    <th class = 'col-sm-2'>Data Criteria</th>
+                    <th class = 'col-sm-3'>Value Set(s)</th>
+                    <th class = 'col-sm-2'>Recorded Code</th>
+                    <th class = 'col-sm-2'>Required Attributes</th>
+                    <th class = 'col-sm-3'>Attribute Details/Value Set</th>
+                    <th class = 'col-sm-2'>Recorded Attribute Value</th>
```
Comment (thecristen) [inline app/views/checklist_tests/_checklist_measures.html.erb]: Bootstrap's based on 12 columns per row but this adds up to 16!!

File: app/views/checklist_tests/_checklist_measures.html.erb [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                <tbody>
+                  <%= f.fields_for :checked_criteria, product_test.checked_criteria.all(measure_id: measure.id) do |criteria_field| %>
+                    <% criteria = measure.hqmf_document[:data_criteria].select { |key, value| key == criteria_field.object.source_data_criteria }.values.first %>
+                    <% if criteria.has_key?('description') %>
+                      <tr>
+                        <td>
+                          <% if criteria_field.object.passed_qrda %>
+                            <i class = 'fa fa-fw fa-check text-success' aria-hidden="true"></i>
+                          <% else %>
+                            <i class = 'fa fa-fw fa-play-circle text-info invisible' aria-hidden="true"></i>
```
Comment (thecristen) [inline app/views/checklist_tests/_checklist_measures.html.erb]: Should add some text to explain the result to screen readers a la 
```
<span class="sr-only">Passes QRDA</span>
```

File: app/views/checklist_tests/_checklist_measures.html.erb [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+              <%= "#{measure.cms_id} #{measure.name}" %>
+            </h1>
+          </div>
+          <div class = 'panel checklist-panel'>
+            <div class = 'panel-body'>
+              <table class = 'table table-sm'>
+                <thead>
+                  <tr>
+                    <th class = 'col-sm-2'>Validated in QRDA</th>
+                    <th class = 'col-sm-2'>Data Criteria</th>
```
Comment (thecristen) [inline app/views/checklist_tests/_checklist_measures.html.erb]: I'm surprised the automated tests aren't yelling at you for this, but `<th>` elements should have an additional attribute describing the scope (e.g. `<th scope="col">asdf</th>`)

File: app/views/checklist_tests/_checklist_measures.html.erb [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+      <div class = 'panel-group' id = '<%= measure.cms_id %>' disabled = true>
+        <div class = 'panel panel-default'>
+          <div class = 'panel-heading'>
+            <h1 class='panel-title'>
+              <%= "#{measure.cms_id} #{measure.name}" %>
+            </h1>
+          </div>
+          <div class = 'panel checklist-panel'>
+            <div class = 'panel-body'>
+              <table class = 'table table-sm'>
```
Comment (thecristen) [inline app/views/checklist_tests/_checklist_measures.html.erb]: `.table-sm` is not really a thing in Bootstrap 3; you might be interested in `.table-condensed`

File: app/helpers/checklist_tests_helper.rb [hunk 9]
```diff
@@ -1,4 +1,13 @@
 module ChecklistTestsHelper
+  def disable_qrda_submission?
+    number_of_criteria = @product_test.checked_criteria.size
+    count = 0
+    @product_test.checked_criteria.each do |criteria|
+      count += 1 if criteria.checklist_complete?
+    end
+    number_of_criteria != count ? true : false
+  end
+
```
Comment (jaebird123) [inline app/helpers/checklist_tests_helper.rb]: function can be `return @product_test.checked_criteria.all? { |criteria| criteria.checklist_complete? }`

File: app/helpers/test_executions_helper.rb [hunk 10]
```diff
@@ -2,8 +2,10 @@ module TestExecutionsHelper
   include TestExecutionsResultsHelper
 
   def displaying_cat1?(task)
-    test = task.product_test
-    (test._type == 'MeasureTest' && task._type == 'C1Task') || (test._type == 'FilteringTest' && task._type == 'Cat1FilterTask')
+    if task
+      test = task.product_test
```
Comment (jaebird123) [inline app/helpers/test_executions_helper.rb]: do not need `test = task.product_test` anymore

File: app/models/checklist_source_data_criteria.rb [hunk 11]
```diff
@@ -11,26 +11,31 @@ class ChecklistSourceDataCriteria
...
+  def checklist_complete?
     if code.blank? && attribute_code.blank? && recorded_result.blank?
       nil
     else
       code_complete != false && attribute_complete != false && result_complete != false
     end
   end
 
+  def complete?
+    checklist_complete? && passed_qrda == true
```
Comment (jaebird123) [inline app/models/checklist_source_data_criteria.rb]: just need `checklist_complete? && passed_qrda`

**Pull request merged**